server/core/terraform/terraform_client.go

```diff
@@ -338,6 +338,11 @@ func (c *DefaultClient) DetectVersion(log logging.SimpleLogging, projectDirector
 	}
 
 	constraint, _ := version.NewConstraint(requiredVersionSetting)
+	// Since terraform version 1.8.2, terraform is not a single file download anymore and
+	// Atlantis fails to download version 1.8.2 and higher. So, as a short-term fix,
+	// we need to block any version higher than 1.8.1 until proper solution is implemented.
+	// More details on the issue here - https://github.com/runatlantis/atlantis/issues/4471
+	highestSupportedConstraint, _ := version.NewConstraint("<= 1.8.1")
```

Comment:
It would be nice if this value was condigurable. I recall a ticket from a frustrated user that could not set the maximum version allowed across their terraform

Reply:
https://github.com/runatlantis/atlantis/issues/3199#issuecomment-1513187272